src/librustc_metadata/rmeta/decoder/cstore_impl.rs

```diff
@@ -128,7 +128,7 @@ provide! { <'tcx> tcx, def_id, other, cdata,
     is_foreign_item => { cdata.is_foreign_item(def_id.index) }
     static_mutability => { cdata.static_mutability(def_id.index) }
     generator_kind => { cdata.generator_kind(def_id.index) }
-    def_kind => { cdata.def_kind(def_id.index) }
+    def_kind => { Some(cdata.def_kind(def_id.index)) }
```

Comment:
What I meant was that you can remove the `Option` from `def_kind` in `query/mod.rs`, btw.